apps/src/lib/levelbuilder/lesson-editor/activitiesEditorRedux.js

```diff
@@ -0,0 +1,555 @@
+import _ from 'lodash';
+
+const INIT = 'activitiesEditor/INIT';
+const ADD_ACTIVITY = 'activitiesEditor/ADD_ACTIVITY';
+const MOVE_ACTIVITY = 'activitiesEditor/MOVE_ACTIVITY';
+const SET_ACTIVITY = 'activitiesEditor/SET_ACTIVITY';
+const REMOVE_ACTIVITY = 'activitiesEditor/REMOVE_ACTIVITY';
+const UPDATE_ACTIVITY_FIELD = 'activitiesEditor/UPDATE_ACTIVITY_FIELD';
+const ADD_ACTIVITY_SECTION = 'activitiesEditor/ADD_ACTIVITY_SECTION';
+const MOVE_ACTIVITY_SECTION = 'activitiesEditor/MOVE_ACTIVITY_SECTION';
+const REMOVE_ACTIVITY_SECTION = 'activitiesEditor/REMOVE_ACTIVITY_SECTION';
+const UPDATE_ACTIVITY_SECTION_FIELD =
+  'activitiesEditor/UPDATE_ACTIVITY_SECTION_FIELD';
+const ADD_TIP = 'activitiesEditor/ADD_TIP';
+const ADD_LEVEL = 'activitiesEditor/ADD_LEVEL';
+const REMOVE_LEVEL = 'activitiesEditor/REMOVE_LEVEL';
+const CHOOSE_LEVEL = 'activitiesEditor/CHOOSE_LEVEL';
+const REORDER_LEVEL = 'activitiesEditor/REORDER_LEVEL';
+const MOVE_LEVEL_TO_ACTIVITY_SECTION =
+  'activitiesEditor/MOVE_LEVEL_TO_ACTIVITY_SECTION';
+const ADD_VARIANT = 'activitiesEditor/ADD_VARIANT';
+const SET_ACTIVE_VARIANT = 'activitiesEditor/SET_ACTIVE_VARIANT';
+const REMOVE_VARIANT = 'activitiesEditor/REMOVE_VARIANT';
+const SET_FIELD = 'activitiesEditor/SET_FIELD';
+const TOGGLE_EXPAND = 'activitiesEditor/TOGGLE_EXPAND';
+
+// NOTE: Position for Activities, Activity Sections and Levels is 1 based.
+
+export const init = (activities, levelKeyList) => ({
```

Comment:
Most / all of these action creators seem like pass-throughs of the arguments, plus a type. Would you be interested in using something like https://redux.js.org/recipes/reducing-boilerplate#generating-action-creators ?

(I'm also not sure what IDEs people are using, and whether this messes up any autocompletion they might offer.)

Reply:
I updated using this. It saves some space but to be honest I don't know how much it really helps.

Reply:
Thanks for trying this out Dani. I don't love the new pattern, because although it does reduce the number of lines, but seems to do so at the expense of added complexity. It might not be the worst pattern to adopt in a small/new codebase, but it doesn't seem compelling enough to start doing, and then either have the codebase be divided on when it is used, and/or make a big effort to standardize on using this.

IMHO it seems neutral in terms of IDE, because (in RubyMine at least) I already have to search for the `type` in order to find where in the reducer we handle a certain action creator.

Maybe we could consider something like this in the future if Redux starts to support this pattern more explicitly and/or IDEs start to understand redux better like they do for React.

By contrast, I really like the idea of making an effort to split large redux components into smaller ones. I think that would be quite valuable here if we can figure out the right way to do it.

Reply:
CC @cforkish who had been spending some cycles thinking about ways to improve how we use Redux in our codebase.

Reply:
Ya I don't love it either. I will revert the change back.